### PR TITLE
Feature search reaper

### LINKED ISSUE #445: Reaper won't read more than one page of webview
### Source information

Reaper scans

### Steps to reproduce

1. Open reaper scans
2. Open any manhwa
3. The list of chapters available will start with the latest 32 as that is how reaper scans lists the manhwa on its site, in pages of 32

### Kotatsu version

6.6.1

### Android version

Android 13

### Other details

Screenshot_2024-01-09-12-00-38-05_a5af3585a93e2016f6667aaf7e1e2f32
Screenshot_2024-01-09-12-00-44-72_a5af3585a93e2016f6667aaf7e1e2f32
Screenshot_2024-01-09-12-00-46-54_a5af3585a93e2016f6667aaf7e1e2f32
Screenshot_2024-01-09-12-00-59-66_a5af3585a93e2016f6667aaf7e1e2f32
Screenshot_2024-01-09-12-01-03-53_a5af3585a93e2016f6667aaf7e1e2f32

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

### LINKED ISSUE #616: Request to add search function in reaperscomics
### Describe your suggested feature

Availability of Search function like in other extensions 

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

### PR DESCRIPTION
**Pull request by NagaYZ**
Added search support for reaperscans.com , and also fixed the issue with only 32 chapters displayed.

Since reaperscans has a rate limit, manga with a lot of chapters will take a bit longer to load.

Closes #445 
Closes #616

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -59,6 +60,7 @@ dependencies {
 	api 'org.jsoup:jsoup:1.17.2'
 	implementation 'org.json:json:20231013'
 	implementation 'androidx.collection:collection:1.4.0'
+	implementation "org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.3"
```
Comment (Koitharu) [inline build.gradle:63]: Is it possible to do this without adding new dependencies? Using org.json?

Comment (NagaYZ) [reply build.gradle]: fixed.

**Comment by Koitharu** [comment]
Thanks!

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/ReaperComics.kt [hunk 2]
```diff
@@ -1,41 +1,81 @@
...
+	override val headers
+		get() = getApiHeaders()
+
+	private val selectTotalChapter = "dl.mt-2 div:nth-child(5) > dd"
+	private val selectState = "dl.mt-2 div:nth-child(4) > dd"
+
+	private val searchCache = mutableSetOf<Manga>() // Cache search results
+	private val chapterCache = mutableMapOf<String, Manga>() // Cache chapter lists
+
+	private val baseUrl = "https://reaperscans.com"
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/ReaperComics.kt:53]: This one should be obtain from `domain` property

**Pull request merged**